src/session_record.js

```
@@ -287,21 +287,30 @@ class SessionRecord {
 
     removeOldSessions() {
-        while (Object.keys(this.sessions).length > CLOSED_SESSIONS_MAX) {
+        const sessionKeys = Object.keys(this.sessions);
+        const sessionsLength = sessionKeys.length;
+
+        for (let i = 0; i < sessionsLength && sessionsLength > CLOSED_SESSIONS_MAX; i++) {
             let oldestKey;
             let oldestSession;
-            for (const [key, session] of Object.entries(this.sessions)) {
+            for (const key of sessionKeys) {
+                if (!key) continue
```

ESLint

[error] Missing semicolon.

(semi)